src/hotspot/share/runtime/os.cpp

```diff
@@ -1668,6 +1668,12 @@ char* os::attempt_reserve_memory_at(char* addr, size_t bytes) {
   char* result = pd_attempt_reserve_memory_at(addr, bytes);
   if (result != NULL) {
     MemTracker::record_virtual_memory_reserve((address)result, bytes, CALLER_PC);
+  } else {
+    LogTarget(Debug, os, metaspace) lt;
+    if (lt.is_enabled()) {
+      lt.print("Attempt to reserve memory at " INTPTR_FORMAT " for "
+                         SIZE_FORMAT " bytes failed\n", p2i(addr), bytes);
+    }
```

Comment:
Can you please add GetLastError? (returns a 32bit unsigned, so use %u and maybe cast to unsigned explicitly: (.. failed (%u)", ..., (unsigned) ::GetLastError())

Is there any reason not to just use log_info(os)("..") instead of LogTarget?

Reply:
With using LogTarget, we can check if it is enabled, and skip the real printing work, but log_info etc also check that I think. I will add to get last error and change to use log_info(os)("....").